tests/b017.py

```diff
@@ -0,0 +1,20 @@
+"""
+Should emit:
+B017 - on lines 10
+"""
+import unittest
+
+
+class Foobar(unittest.TestCase):
+    def evil_raises(self) -> None:
+        with self.assertRaises(Exception):
+            raise Exception("Evil I say!")
+
+    def context_manager_raises(self) -> None:
+        with self.assertRaises(Exception) as ex:
+            raise Exception("Context manager is good")
+        self.assertEqual("Context manager is good", str(ex.exception))
```

Comment:
```suggestion
            raise Exception("Context manager is good")
            self.assertEqual("Context manager is good", str(ex.exception))
```
I feel if this was ran, this would throw a ValueError ? Not important, but yeah.

Reply:
Nope, ex is left in scope when the context manager exits. Been writing my tests like this for years, and Python has never thrown a ValueError.

Reply:
Hmm, that seems to go against the whole context vibe. But sure. I just try not to do it to avoid bugs as I've seen variations of this bite.